stdlib/SuiteSparse/src/umfpack.jl

```diff
@@ -406,6 +426,23 @@ for itype in UmfpackIndexTypes
                         mx, mz, C_NULL, lu.numeric, umf_info)
             complex(mx[], mz[])
         end
+        function logabsdet(F::UmfpackLU{T, $itype}) where {T} # return log(abs(det)) and sign(det)
+            n = checksquare(F)
+            issuccess(F) || return log(zero(real(T))), log(one(T))
+            U = F.U
+            Rs = F.Rs
+            p = F.p
+            q = F.q
+            s = _signperm(p)*_signperm(q)*one(real(T))
+            P = one(T)
+            abs_det = zero(real(T))
+            @inbounds for i in 1:n
+                dg_ii = U[i, i] / Rs[i]
+                P *= sign(dg_ii)
+                abs_det += log(abs(dg_ii))
```

Comment:
It might be slightly faster (and more accurate) to use `log2` in the sum, and multiply by `1/log(2)` at the end

Reply:
Just tried this locally, and it doesn't seem to make any difference to timing (n.b., division at the end is by `log2(e)`, not `log(2)`).  Given that, I think leaving everything in base-e is probably clearer?